libs/entitlement/qa/src/javaRestTest/java/org/elasticsearch/entitlement/qa/EntitlementsTestRule.java

```diff
@@ -38,7 +38,7 @@ class EntitlementsTestRule implements TestRule {
             Map.of(
                 "files",
                 List.of(
-                    Map.of("path", tempDir.resolve("read_dir"), "mode", "read"),
+                    Map.of("path", tempDir.resolve("read_dir"), "mode", "read_write"),
```

Comment:
The Entitled plugin needs read-write permission on the read_dir, so it can expose a function to create files into the read-only dir (and we can test we are allowed or not to read them)